configs/components/ruby-2.4.2.rb

```diff
@@ -0,0 +1,302 @@
+component "ruby-2.4.2" do |pkg, settings, platform|
+  pkg.version "2.4.2"
+  pkg.md5sum "fe106eed9738c4e03813ab904f8d891c"
+  pkg.url "https://cache.ruby-lang.org/pub/ruby/2.4/ruby-#{pkg.get_version}.tar.gz"
+  pkg.mirror "#{settings[:buildsources_url]}/ruby-#{pkg.get_version}.tar.gz"
+
```

Comment:
I think you also need to add this part:

https://github.com/puppetlabs/puppet-agent/blob/master/configs/components/ruby-2.4.2.rb#L7-L11

Reply:
There are a whole lot of places these components will need to be modified for other platforms; I haven't really begun that part yet. My plan for this work was something along these lines:

- Make the existing components in puppet-runtime work for all agent branches on one platform first
- Add any new components that make sense from the puppet-agent repo (e.g. other ruby gems if possible, etc.)
- Add support for other platforms